File: Problem Solving/Grading Students/code.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'gradingStudents' function below.
#
# The function is expected to return an INTEGER_ARRAY.
# The function accepts INTEGER_ARRAY grades as parameter.
#

def gradingStudents(grades):
    roundgrades = []
    
    for i in grades:
        j = i
        if j>37:
            if (j%5)>2:
                j+=5-(j%5)
                roundgrades.append(j)
            else:
                roundgrades.append(j)
        else:
            roundgrades.append(j)
    
    return roundgrades
    # Write your code here

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    grades_count = int(input().strip())

    grades = []

    for _ in range(grades_count):
        grades_item = int(input().strip())
        grades.append(grades_item)

    result = gradingStudents(grades)

    fptr.write('\n'.join(map(str, result)))
    fptr.write('\n')

    fptr.close()

File: Problem Solving/Alternating Characters/code.cpp
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the alternatingCharacters function below.
def alternatingCharacters(s):
    S = []
    T = []
    for i in range(len(s)):
        S.append(s[i])
    print(S)

    j=0
    i=0

    while i<len(S):
        if len(S)>1:
            try:
                while S[i] == S[i+1]:
                    del S[i+1]
                    j+=1

            except IndexError:
                break
        i+=1
        
    return j

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    q = int(input())

    for q_itr in range(q):
        s = input()

        result = alternatingCharacters(s)
        fptr.write(str(result) + '\n')

    fptr.close()
